cdk/lib/atom-preview.ts

```diff
@@ -1,11 +1,46 @@
-import { join } from "path";
-import { CfnInclude } from "@aws-cdk/cloudformation-include";
 import type { App } from "@aws-cdk/core";
 import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
-import { GuStack, GuStageParameter } from "@guardian/cdk/lib/constructs/core";
+import { GuStack } from "@guardian/cdk/lib/constructs/core";
+import { GuEc2App} from "@guardian/cdk";
+import {AccessScope, Stage} from "@guardian/cdk/lib/constants";
+import {InstanceClass, InstanceSize, InstanceType} from "@aws-cdk/aws-ec2";
+import {GuCname} from "@guardian/cdk/lib/constructs/dns";
+import {Duration} from "@aws-cdk/core";
 
 export class AtomPreview extends GuStack {
   constructor(scope: App, id: string, props: GuStackProps) {
     super(scope, id, props);
+    const { loadBalancer } = new GuEc2App(this, {
+      applicationPort: 9000,
+      app: "atom-preview",
+      access: { scope: AccessScope.PUBLIC },
+      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.SMALL),
+      certificateProps:{
+        [Stage.CODE]: {
+          domainName: "atom-preview.code.dev-gutools.co.uk",
+        },
+        [Stage.PROD]: {
+          domainName: "atom-preview.gutools.co.uk",
+        },
+      },
+      monitoringConfiguration: {
+        noMonitoring: true,
+      },
+      userData: {
+        distributable: {
+          fileName: "atom-preview.deb",
+          executionStatement: `dpkg -i /atom-preview/atom-preview.deb`,
+        }
+      },
+      scaling: {
+        [Stage.CODE]: {
+          minimumInstances: 1
+        },
+        [Stage.PROD]: {
+          minimumInstances: 1
+        }
+      }
+    });
+    new GuCname(this, "dns", {app: "atom-preview", domainName:  "atom-preview.code.dev-gutools.co.uk", resourceRecord: loadBalancer.loadBalancerDnsName, ttl: Duration.hours(1),})
```

Comment:
The value of `domainName` works well for CODE, but not PROD. Not a blocker for this PR though, and we should move to [v39+](https://github.com/guardian/cdk/releases/tag/v39.0.0) which vastly simplifies stage awareness.